src/PotentialSimulation/ScalarPotential.jl

```diff
@@ -16,76 +20,79 @@ function getindex(ep::P, g::Grid{T, N, S}) where {T, N, S, P <: ScalarPotential{
 end
 
 
-function get_2π_potential(wp::ScalarPotential{T, 3, Cylindrical}, axφ::DiscreteAxis{T, :periodic, :periodic}, int::Interval{:closed, :open, T}) where {T}
+function get_2π_potential(sp::ScalarPotential{T, 3, Cylindrical}, axφ::DiscreteAxis{T, :periodic, :periodic}, int::Interval{:closed, :open, T}) where {T}
     @assert int.right != 0 "Right boundary of φ interval is not allowed to be 0"
-    Potential::Type = typeof(wp)
     l::Int = length( axφ )
     Δφ::T = int.right - int.left
     new_int::Interval{:closed, :open, T} = Interval{:closed, :open, T}(0, 2π)
     n::Int = Int(round(T(2π) / Δφ, sigdigits = 6))
     new_ticks::Vector{T} = Vector{T}(undef, l * n)
-    new_pot::Array{T, 3} = Array{T, 3}(undef, size(wp, 1), l * n, size(wp, 3))
+    new_pot = Array{eltype(sp.data), 3}(undef, size(sp, 1), l * n, size(sp, 3))
```

Comment:
Why use `eltype(sp.data)` instead of `T`?

Reply:
Because it is not the same. `PointTypes` for example have `T = UInt8` whereas the grid is usually in e.g. `AT = Float32`.
Its better solved now.